mlflow/sklearn/utils.py

```diff
@@ -310,19 +312,24 @@ def plot_confusion_matrix(*args, **kwargs):
                 "axes.labelsize": 10.0,
             }
         ):
-            return sklearn.metrics.plot_confusion_matrix(*args, **kwargs)
+            return (
+                sklearn.metrics.ConfusionMatrixDisplay.from_estimator(*args, **kwargs)
+                if is_plot_function_deprecated
+                else sklearn.metrics.plot_confusion_matrix(*args, **kwargs)
+            )
 
+    target_label_arg_name = "y" if is_plot_function_deprecated else "y_true"
```

Comment:
@shaikmoeed `sklearn.metrics.ConfusionMatrixDisplay.from_estimator` uses `y` for target labels but `sklearn.metrics.plot_confusion_matrix` uses `y_true`.

Reply:
This should fix the `Cross version tests / test (sklearn / 1.0.2 / autologging)` failure.

Reply:
Thanks for making changes and explaining them. I got a better picture/learning to contribute next time. I hope, all the test cases run successfully :)